tests/reference.yaml

```diff
@@ -1054,6 +1054,9 @@ definitions:
         type: string
         format: date
         readOnly: true
+      owner:
+        type: string
+        default: CurrentUserDefault()
```

Comment:
I don't see how this is appropriate. 

The string `CurrentUserDefault()` does not represent a valid user ID, so cannot be a "default value" for the owner field which is a PrimaryKeyRelatedField.

As far as I know, the OpenAPI specification provides no way to specify a default based on external state such as currently authenticated user, so this can't really be expressed by spec. The best you could hope for would be to document the behavior in a description, e.g. "if `owner` is not provided, it will be assumed to be the currently logged in user".

Since this project aims for spec compliance and codegen tool compatibility, we cannot just use arbitrary strings just for documentation purposes in the `default` field.

Reply:
In fact, if [this TODO](https://github.com/axnsan12/drf-yasg/blob/7683a28816f85478fda96a60eb2fa2fe10bc4c38/src/drf_yasg/utils.py#L242) were to be implemented, the owner field would be inferred to be of type `int` (like `pk` of `User`), and "CurrentUserDefault()" would not even conform to the type.

According to [the OpenAPI spec](https://github.com/axnsan12/drf-yasg/blob/7683a28816f85478fda96a60eb2fa2fe10bc4c38/src/drf_yasg/utils.py#L242), the value of `default` must conform to the `Schema` it is declared in.